types/lru-cache/index.d.ts

```diff
@@ -1,92 +1,93 @@
+     * The total number of items held in the cache at the current moment.
      */
-    readonly itemCount: number;
+    readonly size: number;
 
     /**
-     * Same as Options.allowStale.
+     * The total size of items in cache when using size tracking.
      */
-    allowStale: boolean;
+    readonly calculatedSize: number;
 
     /**
-     * Same as Options.length.
+     * Add a value to the cache.
      */
-    lengthCalculator(value: V): number;
+    set(key: K, value: V, options?: LRUCache.SetOptions<K, V>): this;
 
     /**
-     * Same as Options.max. Resizes the cache when the `max` changes.
+     * Return a value from the cache.
+     * Will update the recency of the cache entry found.
+     * If the key is not found, `get()` will return `undefined`.
+     * This can be confusing when setting values specifically to `undefined`,
+     * as in `cache.set(key, undefined)`. Use `cache.has()` to determine
+     * whether a key is present in the cache at all.
      */
-    max: number;
+    get(key: K, options?: LRUCache.GetOptions): V | undefined;
 
     /**
-     * Same as Options.maxAge. Resizes the cache when the `maxAge` changes.
+     * Like `get()` but doesn't update recency or delete stale items.
+     * Returns `undefined` if the item is stale, unless `allowStale` is set either on the cache or in the options object.
      */
-    maxAge: number;
+    peek(key: K, options?: LRUCache.PeekOptions): V | undefined;
 
     /**
-     * Will update the "recently used"-ness of the key. They do what you think.
-     * `maxAge` is optional and overrides the cache `maxAge` option if provided.
+     * Check if a key is in the cache, without updating the recency or age.
+     * Will return false if the item is stale, even though it is technically in the cache.
      */
-    set(key: K, value: V, maxAge?: number): boolean;
+    has(key: K): boolean;
 
     /**
-     * Will update the "recently used"-ness of the key. They do what you think.
-     * `maxAge` is optional and overrides the cache `maxAge` option if provided.
-     *
-     * If the key is not found, will return `undefined`.
+     * Deletes a key out of the cache.
+     * Returns true if the key was deleted, false otherwise.
      */
-    get(key: K): V | undefined;
+    delete(key: K): boolean;
 
     /**
-     * Returns the key value (or `undefined` if not found) without updating
-     * the "recently used"-ness of the key.
-     *
-     * (If you find yourself using this a lot, you might be using the wrong
-     * sort of data structure, but there are some use cases where it's handy.)
+     * Clear the cache entirely, throwing away all values.
      */
-    peek(key: K): V | undefined;
+    clear(): void;
 
-    /**
-     * Check if a key is in the cache, without updating the recent-ness
-     * or deleting it for being stale.
-     */
-    has(key: K): boolean;
+    /** @deprecated */
+    reset(): void;
```

Comment:
hi @dhensby @andrewbranch
I added more descriptions for deprecated definitions, utilize work from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58853, thanks to @martin-badin

```ts
    /**
     * Clear the cache entirely, throwing away all values.
     *
     * @deprecated use clear() instead
     * @since 7.0.0
     */
    reset(): void;
```